x-pack/plugins/observability/public/hooks/create_use_rules_link.ts

```diff
@@ -6,17 +6,13 @@
  */
 import { Options, useLinkProps } from './use_link_props';
 
-export function createUseRulesLink(isNewRuleManagementEnabled = false) {
-  return function (options: Options = {}) {
-    const linkProps = isNewRuleManagementEnabled
-      ? {
-          app: 'observability',
-          pathname: '/alerts/rules',
-        }
-      : {
-          app: 'management',
-          pathname: '/insightsAndAlerting/triggersActions/rules',
-        };
+export const createUseRulesLink =
```

Comment:
@mgiota, do we still need this hook?

Reply:
@fkanout Currently it is being used by logs, metrics, apm and uptime. The ultimate goal was to re-use it even within observability, but I remember I had some issues and I didn't use it in the end. So I would say it is good to still keep it.

Reply:
Here's a bit of context of how things were before https://github.com/elastic/kibana/pull/124486. There was already a hook that was used by logs and metrics and what I've done in the above mentioned PR was to simply move the hook in the observability plugin, so that it can be reused by apm and uptime as well. This way I didn't have to repeat the observability rules link construction logic (use stack management url if rules are disabled, otherwise use o11y rules link) in all consumers, just in one place, in the hook. If you notice at this point (in the PR I linked above) the hook was called useRulesLink. Then I improved things more https://github.com/elastic/kibana/pull/124132/commits/fad5ff115ee79a62da641726e95bf9c88c93657d and eventually I ended up having a useRulesLink hook that uses a closure (`createUseRulesLink `) that contained the config information I needed. 

Here's how uptime was before and after the introduction of the hook https://github.com/elastic/kibana/pull/124132/commits/44bb2fd8d8625ca1dde3070785824b2e2249e0f1#diff-21c308974453586a210671ef718e5d2e2b4c9222e2d1888d227b1508c9937df9R109. If we didn't have the hook now, I would need to do changes in all consumers and yes in this case you should see changes in this PR related to these apps. The beauty of the hook is that you don't see any changes on the consumer side :-)  Here's another example use case for [metrics](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/alerting/common/components/metrics_alert_dropdown.tsx#L89) after I moved the hook to the observability.

Reply:
Since I don't need to access the config anymore, what I could further improve in this PR though is following:

- rename `create_use_rules_link` to `use_rules_link`
- `export { useRulesLink } from './hooks/use_rules_link';` instead of returning it here `kibana/x-pack/plugins/observability/public/plugin.ts`

The only way to access the config information from observability app from rest apps was only through the closure I created (createUseRulesLink). Now that we don't need the config information, I can do the refactoring I mentioned above. Do you want me to proceed with the refactoring?

Reply:
@fkanout Here's the refactoring I am talking about https://github.com/mgiota/kibana/commit/94a3604dc6eff634fdd0cc9c5d5bebdaa302ae67. I am just getting an eslint error. I'll have a look how I can fix. Hopefully it won't take much. Otherwise I don't want to block this PR

Reply:
@mgiota, thanks for the explanation. 
>rename create_use_rules_link back to use_rules_link and update accordingly

Definitely a good idea because the naming is confusing. `createUseRulesLink ` vs `useRulesLink`
However, I think a better idea would be to delete the hook as there is no more logic inside it (switch between URLs) and do it like this inside the `plugin.ts`:
```
 useRulesLink: useLinkProps({
        app: 'observability',
        pathname: '/alerts/rules',
      }),
```
Do you agree that will do the job?

Reply:
Regarding the refactor here https://github.com/elastic/kibana/pull/134427, I frankly think it's unnecessary. It potentially would increase the bundle as you are adding import in every place this rule's link page is required. While with the current implementation we have it for free from the `kibana -> services -> observability`.

Reply:
> ```
>  useRulesLink: useLinkProps({
>         app: 'observability',
>         pathname: '/alerts/rules',
>       }),
> ```
> 
> Do you agree that will do the job?

Unfortunately this wouldn't work. Here's the error I get in Visual Studio and the app doesn't even start.

<img width="868" alt="Screenshot 2022-06-15 at 15 02 24" src="https://user-images.githubusercontent.com/2852703/173833969-83ca2d2b-a53d-4e83-bdee-b8c9efeb5194.png">


> However, I think a better idea would be to delete the hook as there is no more logic inside it (switch between URLs)

I don't completely agree. True it doesn't contain the logic to switch between URLs, but it contains the rules link url. This was the addition to the previous `useLinkProps` hook (encapsulate the rules link to not have to repeat it when consuming the hook) https://github.com/elastic/kibana/pull/124486/files#diff-515673b6758813063920def6cab40ecb8bf6e00b77c57c7613931febafe7942eR86. Adding more logic to the hook (some decision based on configuration) was a next addition.

 
> Regarding the refactor here https://github.com/elastic/kibana/pull/134427, I frankly think it's unnecessary.

Ok I removed the refactoring from current PR, since this whole topic I think goes beyond scope of this PR. Could we tackle this in a separate PR?

Reply:
@mgiota, ok. Let's keep the hook for now, and remove the refactoring here https://github.com/elastic/kibana/pull/134427.

Reply:
Ok! I'm putting here a suggestion by Milton https://github.com/elastic/kibana/pull/124486#issuecomment-1028975245 on how we could improve things further